contrib/runsim/notification.go

```diff
@@ -1,10 +1,97 @@
 package main
 
 import (
-	"github.com/nlopes/slack"
+	"fmt"
 	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/nlopes/slack"
+)
+
+const (
+	logBucketPrefix = "sim-logs-"
+	awsRegion       = "us-east-1"
+)
+
+var (
+	simTimeStamp = time.Now().Format("01-02-2006_15:05:05")
 )
 
+func awsErrHandler(err error) {
+	if awsErr, ok := err.(awserr.Error); ok {
+		switch awsErr.Code() {
+		default:
+			log.Println(awsErr.Error())
+		}
+	} else {
+		log.Println(err.Error())
+	}
+}
+
+func makeObjKey(folderName string, fileName string) string {
+	return fmt.Sprintf("%s/%s/%s", folderName, simTimeStamp, fileName)
+}
+
+func putObj(fileHandle *os.File, svc *s3.S3, folderName string, bucketName string) {
+	_, _ = fileHandle.Seek(0, 0)
+
+	stdOutObjInput := &s3.PutObjectInput{
+		Body:   aws.ReadSeekCloser(fileHandle),
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(makeObjKey(folderName, filepath.Base(fileHandle.Name()))),
+	}
+	if output, err := svc.PutObject(stdOutObjInput); err != nil {
+		awsErrHandler(err)
+	} else {
+		log.Printf("Log file pushed: %s", output.String())
+	}
+}
+
+func pushLogs(stdOut *os.File, stdErr *os.File, folderName string) {
+	var logBucket *string
+
+	sessionS3 := s3.New(session.Must(session.NewSession(&aws.Config{
+		Region: aws.String(awsRegion),
+	})))
+	if listBucketsOutput, err := sessionS3.ListBuckets(&s3.ListBucketsInput{}); err != nil {
+		awsErrHandler(err)
+	} else {
```

Comment:
For readability's sake I'd remove this `else` by returning after the `awsErrHandler()` call.

Reply:
Yeah, I was going to do something more in that else block but didn't and forgot to take it out. I'll change it in the new repo if that's cool with you.